CaImAn/caiman_main.py

```diff
@@ -0,0 +1,316 @@
+
+
+
+def preview(caiman_parameters: cm_params.CaimanParameters):
+    """
+    in HDF5 file
+    """
+
+    # Import for CaimAn lib
+    from summary_images import correlation_pnr
+
+    video_start_frame, video_stop_frame   = caiman_parameters.preview_parameters[defs.Parameters.Preview.RANGE]
+
+    with h5py.File(caiman_parameters.paths[defs.Parameters.Path.FILEPATH], 'r') as file_:
+        if defs.DoricFile.Dataset.IMAGE_STACK in file_[caiman_parameters.paths[defs.Parameters.Path.H5PATH]]:
+            images = np.array(file_[f"{caiman_parameters.paths[defs.Parameters.Path.H5PATH]}/{defs.DoricFile.Dataset.IMAGE_STACK}"])
+        else:
+            images = np.array(file_[f"{caiman_parameters.paths[defs.Parameters.Path.H5PATH]}/{defs.DoricFile.Deprecated.Dataset.IMAGES_STACK}"])
+
+    images = images[:, :, (video_start_frame-1):video_stop_frame]
+    images = images[:, :, ::caiman_parameters.preview_parameters[defs.Parameters.Preview.TEMPORAL_DOWNSAMPLE]]
+
+    images = images.transpose(2, 0, 1)
+
+    cr, pnr = correlation_pnr(images, swap_dim = False)
+
+    cr[np.isnan(cr)] = 0
+    pnr[np.isnan(pnr)] = 0
+
+    try:
+        with h5py.File(caiman_parameters.preview_parameters[defs.Parameters.Preview.FILEPATH], 'w') as hdf5_file:
+            hdf5_file.create_dataset(cm_defs.Preview.Dataset.LOCALCORR, data = cr, dtype = "float64", chunks = True)
+            hdf5_file.create_dataset(cm_defs.Preview.Dataset.PN, data = pnr, dtype = "float64", chunks = True)
+
+    except Exception as error:
+        utils.print_error(error, cm_defs.Messages.SAVE_TO_HDF5)
+
+
+def save_caiman_to_doric(
+    Y: np.ndarray,
+    A: np.ndarray,
+    C: np.ndarray,
+    S: np.ndarray,
+    shape,
+    time_: np.array,
+    bit_count: int,
+    qt_format: int,
+    username: str,
+    vname: str = "caiman.doric",
+    vpath: str = "DataProcessed/MicroscopeDriver-1stGen1C",
+    vdataset: str = "Series1/Sensor1",
+    params_doric: Optional[dict] = {},
+    params_source: Optional[dict] = {},
+    saveimages: bool = True,
+    saveresiduals: bool = True,
+    savespikes: bool = True
+) -> str:
+    """
+    Save CaImAn results to .doric file:
+
+    Since the CNMF algorithm contains various arbitrary scaling process, a normalizing
+    scalar is computed with least square using a subset of frames from `Y` and `AC`
+    such that their numerical values matches.
+
+    """
+
+    vpath    = utils.clean_path(vpath)
+    vdataset = utils.clean_path(vdataset)
+
+    AC = A.dot(C)
+    res = Y - AC
+
+    AC = AC.reshape(shape, order='F').transpose((-1, 0, 1))
+    res = res.reshape(shape, order='F').transpose((-1, 0, 1))
+    A = A.toarray()
+    A = A.reshape(shape[0],shape[1],A.shape[1], order='F').transpose((-1, 0, 1))
+
+    print(cm_defs.Messages.GEN_ROI_NAMES, flush = True)
+    names = []
+    roiUsernames = []
+    for i in range(len(C)):
+        names.append("ROI"+str(i+1).zfill(4))
+        roiUsernames.append(f"ROI {i+1}")
```

Comment:
we are generating the ROI names and usernames here (same for minian) but we are not really using them only when saving the spikes.

In the function save_roi we are generating them too so maybe I should give them to s`ave_roi_signals` or get them from `save_roi_signals`